modules/claims_api/app/controllers/concerns/claims_api/poa_verification.rb

```diff
@@ -81,11 +81,24 @@ def verify_representative_and_veteran(logged_in_representative_user, target_vete
       end
 
       def poa_code_in_organization?(poa_code)
+        organization = ::Veteran::Service::Organization.find_by(poa: poa_code)
+
+        if user_is_target_veteran?
+          organization.present?
+        else
+          representative = ::Veteran::Service::Representative.find_by(first_name: @current_user.first_name,
+                                                                      last_name: @current_user.last_name)
+          organization.present? && representative.poa_codes.include?(poa_code)
+        end
```

Comment:
I'm not sure if it's because this PR is too big, or if there's too many different pieces being changed at once, but this logic concerns me.  It feels like a method named, `poa_code_in_organization?` should only be checking to see if the given POA code belongs to an organization and that's it.  It feels like if we need to also make sure that the authenticated person is allowed to use that POA code, then that should be an entirely separate check outside of this logic.

Reply:
>`poa_code_in_organization?` should only be checking to see if the given POA code belongs to an organization and that's it.

I agree. I think this is more of a poorly named method, since the [existing method](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/claims_api/app/controllers/concerns/claims_api/poa_verification.rb#L83) does more than that, as well. Or rather, it does it in the roundabout way, i.e., "is there a representative with this POA code _and_ do they have `veteran_service_officer` user type? 

I would propose instead that we refactor the existing method to be much smaller and move the rest of the logic downstream into the controller, which would be clearer. E.g.
```ruby
def poa_code_in_organization?(poa_code)
   ::Veteran::Service::Organization.find_by(poa: poa_code).present?
end
```
then, we're really addressing 2 scenarios, which right now is the `if-else` clause above:
1. If the request is being made by a veteran, is the POA code for an organization?
2. If the request is being made by a representative, is the POA for an organization **AND** is the rep authorized to act on behalf of that organization

This gets rid of some of the version-specific gaps I had to bridge with adding `user_is_target_veteran?` to v1. now that could just be a header_request?` check in the v1 controller and a `user_is_target_veteran?` check in v2. 

What are your thoughts around that approach?

Reply:
Yeah, I'm all for making things simpler.  So if we need to go back to the existing method and clean it up so that we can make everything make more sense, then let's definitely do that.